Introducing the node settings views

jarolrod opened the pull request:
This introduces the views that would allow for [settings configuration](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App?node-id=4588%3A101966&t=m6V59cZ2STeCRp1y-0) after the user has finished onboarding and the node has started to run.

| NodeRuner | NodeSettings |
| --------- | ------------ |
| <img width="752" alt="Screen Shot 2022-12-10 at 12 58 07 AM" src="https://user-images.githubusercontent.com/23396902/206832668-0a5e469a-830f-4d7e-bfc5-0e6c115d74fd.png"> | <img width="752" alt="Screen Shot 2022-12-10 at 12 58 12 AM" src="https://user-images.githubusercontent.com/23396902/206832715-299234d4-6850-4dd3-956f-fd5bc47980c7.png"> |

For several reasons this also entangles the About and Developer options and entangles them into their own independent SwipeView, which means that we only need to instantiate SettingsAbout. There isn't a use case for the Developer options to be shown independently.

### Follow-ups:
- The caret-right icon used within the settings rows is too large because I'm just using NavButton, the width and height should be adjusted to match closer to what the design file [specifies](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App?node-id=4588%3A101966&t=m6V59cZ2STeCRp1y-0)
- This uses a StackView to move through the settings pages but on linux, the animation between pages isn't how we'd want it to be; it's fine on macOS. The animation should match our SwipeViews. Note: I didn't use SwipeView because while you can set an index to go to, it flips through all previous pages to get there which is visually distracting.



[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/203)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/203)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/203)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/203)

hebasto commented:
Rebase needed.

jarolrod commented:
updated from e7763aa to 942a3b4

changes: rebased over master

johnny9 commented on src/qml/pages/node/NodeSettings.qml:
```diff
@@ -0,0 +1,114 @@
+        initialItem: node_settings
+        anchors.fill: parent
+    }
+    Page {
```
This should be defined as a Component so it isn't loaded twice.

johnny9 commented on src/qml/pages/node/NodeSettings.qml:
```diff
@@ -0,0 +1,114 @@
+import "../../components"
+import "../settings"
+
+Page {
```
Can just be an Item since header and footer are not set at this level.

johnny9 commented on src/qml/pages/node/NodeSettings.qml:
```diff
@@ -0,0 +1,114 @@
+        ColumnLayout {
+            spacing: 0
+            width: parent.width
+            ColumnLayout {
```
Seems like one of these ColumnLayouts can be removed

jarolrod replied:
this allows for the table to expand but be a max width of 450, we use the same nested approach in the `InformationPage` control; open to better ways of accomplishing this

johnny9 replied:
Can use width: Math.min(parent.width, 450)

johnny9 commented on src/qml/pages/settings/SettingsAbout.qml:
```diff
@@ -8,19 +8,43 @@ import QtQuick.Layouts 1.15
-        AboutOptions {
-            Layout.maximumWidth: 450
-            Layout.alignment: Qt.AlignCenter
+Page {
```
This can be Item since header/footer are not set.

jarolrod commented:
updated from 942a3b4 to 06235ca ([pr203.02](https://github.com/jarolrod/gui-qml/tree/pr203.02) -> [pr203.03](https://github.com/jarolrod/gui-qml/tree/pr203.03), [diff](https://github.com/jarolrod/gui-qml/compare/pr203.02..pr203.03))

changes: addressed @johnny9 suggestions -1

jarolrod commented:
updated from 942a3b4 to 06235ca ([pr203.03](https://github.com/jarolrod/gui-qml/tree/pr203.03) -> [pr203.04](https://github.com/jarolrod/gui-qml/tree/pr203.04), [diff](https://github.com/jarolrod/gui-qml/compare/pr203.03..pr203.04))

changes: rebased over changes on master

johnny9 commented on src/qml/pages/node/NodeRunner.qml:
```diff
@@ -0,0 +1,43 @@
+    ColumnLayout {
+        spacing: 0
+        anchors.fill: parent
+        ColumnLayout {
```
Here's another double ColumnLayout.  I think we should reduce the layers here or use a different top level Item.

jarolrod replied:
maybe we can leave it to a follow-up pr to address the nested columnlayouts throughout the code?

johnny9 reviewed (approved):
ACK 6037471dbbe88a4c34daefc0095a7d183af2b38d

Changes look good. Built and ran with not qml warnings/errors and views render as expected. Will address ColumnLayout comments in a follow up PR.
